Std/Lean/Util/EnvSearch.lean

```diff
@@ -0,0 +1,45 @@
+/-
+Copyright (c) 2021 Shing Tak Lam. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Shing Tak Lam, Daniel Selsam, Mario Carneiro
+-/
+
+import Lean.Modifiers
+
+open Lean
+
+namespace Lean
+
+/--
+Options to control getMatchingConstants options below.
+-/
+structure EnvironmentSearchOptions where
+  /-- Include declarations in imported environment. -/
+  stage1       : Bool := true
+  /-- Include private declarations. -/
+  checkPrivate : Bool := false
+
+/--
+Find constants in current environment that match find options and predicate.
+-/
+def getMatchingConstants {m} [Monad m] [MonadEnv m]
+    (p : ConstantInfo → m Bool)
+    (opts : EnvironmentSearchOptions := {})
+    : m (Array ConstantInfo) := do
+  let matches_ ← if opts.stage1 then
+                    (← getEnv).constants.map₁.foldM (init := #[]) check
+                   else
+                    pure #[]
```

Comment:
I would have formatted this as 
```suggestion
  let matches_ ← 
    if opts.stage1 then
      (← getEnv).constants.map₁.foldM (init := #[]) check
    else
      pure #[]
```
but I'm not sure there is actually a style rule against deep indenting like this?